presto-accumulo/src/main/java/com/facebook/presto/accumulo/io/AccumuloPageSink.java

```diff
@@ -0,0 +1,292 @@
+import org.apache.accumulo.core.client.BatchWriterConfig;
+import org.apache.accumulo.core.client.Connector;
+import org.apache.accumulo.core.client.MutationsRejectedException;
+import org.apache.accumulo.core.client.TableNotFoundException;
+import org.apache.accumulo.core.data.Mutation;
+import org.apache.accumulo.core.data.Value;
+import org.apache.hadoop.io.Text;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import static com.facebook.presto.accumulo.AccumuloErrorCode.INTERNAL_ERROR;
+import static com.facebook.presto.accumulo.AccumuloErrorCode.UNEXPECTED_ACCUMULO_ERROR;
+import static com.facebook.presto.accumulo.AccumuloErrorCode.VALIDATION;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Output class for serializing Presto pages (blocks of rows of data) to Accumulo. This class
+ * converts the rows from within a page to a collection of Accumulo Mutations, writing and indexed
+ * the rows. Writers are flushed and closed on commit, and if a rollback occurs... we'll you're
+ * gonna have a bad time.
+ *
+ * @see AccumuloPageSinkProvider
+ */
+public class AccumuloPageSink
+        implements ConnectorPageSink
+{
+    private final AccumuloRowSerializer serializer;
+    private final BatchWriter wrtr;
+    private final Indexer indexer;
+    private final List<AccumuloColumnHandle> columns;
+    private Integer rowIdOrdinal;
+
+    /**
+     * Creates a new instance of {@link AccumuloPageSink}
+     *
+     * @param config Configuration for Accumulo
+     * @param table Table metadata for an Accumulo table
+     */
+    public AccumuloPageSink(AccumuloConfig config, AccumuloTable table)
+    {
+        requireNonNull(config, "config is null");
+        requireNonNull(table, "table is null");
+
+        this.columns = table.getColumns();
+
+        // Fetch the row ID ordinal, throwing an exception if not found for safety
+        for (AccumuloColumnHandle ach : columns) {
+            if (ach.getName().equals(table.getRowId())) {
+                rowIdOrdinal = ach.getOrdinal();
+                break;
+            }
+        }
+
+        if (rowIdOrdinal == null) {
+            throw new PrestoException(INTERNAL_ERROR, "Row ID ordinal not found");
+        }
+
+        this.serializer = table.getSerializerInstance();
+
+        try {
+            Connector conn = AccumuloClient.getAccumuloConnector(config);
+            // Create a BatchWriter to the Accumulo table
+            BatchWriterConfig conf = new BatchWriterConfig();
+            wrtr = conn.createBatchWriter(table.getFullTableName(), conf);
+
+            // If the table is indexed, create an instance of an Indexer, else null
+            if (table.isIndexed()) {
+                indexer = new Indexer(conn,
+                        conn.securityOperations().getUserAuthorizations(config.getUsername()),
+                        table, conf);
+            }
+            else {
+                indexer = null;
```

Comment:
Make `indexer` `Optional` to indicate there might be no indexer instead of depending on `null` checks.